src/binder/bind/bind_graph_pattern.cpp

```diff
@@ -327,10 +327,20 @@ std::shared_ptr<RelExpression> Binder::createRecursiveQueryRel(const parser::Rel
 std::pair<uint64_t, uint64_t> Binder::bindVariableLengthRelBound(
     const kuzu::parser::RelPattern& relPattern) {
     auto recursiveInfo = relPattern.getRecursiveInfo();
-    auto lowerBound = std::min(TypeUtils::convertToUint32(recursiveInfo->lowerBound.c_str()),
-        clientContext->varLengthExtendMaxDepth);
-    auto upperBound = std::min(TypeUtils::convertToUint32(recursiveInfo->upperBound.c_str()),
-        clientContext->varLengthExtendMaxDepth);
+    auto lowerBound = TypeUtils::convertToUint32(recursiveInfo->lowerBound.c_str());
+    auto upperBound = clientContext->varLengthExtendMaxDepth;
+    if (!recursiveInfo->upperBound.empty()) {
+        upperBound = TypeUtils::convertToUint32(recursiveInfo->upperBound.c_str());
+    }
+
```

Comment:
Remove empty line.

You are now checking lowerBound <= limit then upperBound <= limit.

I guess you can just check lowerBound <= upperBound and then upperBound <= limit. This just comes to my mind that we didn't validate lowerBound <= upperBound.